### DISCUSSION TITLE
How to specify the parent local interface for routing?

### DISCUSSION THREAD

**Discussion by zstewart93** [qdm12/gluetun#834] [Q&A] [answered]
I wasn't seeing this as an environment variable but I have several interfaces on the server which are bridged into a "bond0" interface which is the only interface that routes out to my local network. Upon starting the container everything looks good and it seems to have downloaded the config files and has internet access in the container, but the log says it's "found eth0" and is using that for routing. However that interface is not the "correct" one. As a result I cannot use shadowsocks or httpproxy or hit the port 8000 page on any of my local machines.

Any thoughts?

**Comment by qdm12** [comment]
Uh interesting, if you run `docker run -it --rm alpine:3.15 ip link` what do you get?

I could change the code to be able to manually specify an interface name if you think there is no way to auto-detect which one to use. Let me know and I'll create an issue for it (or feel free to do it too).

EDIT: How about allowing inbound traffic from all the local interfaces?

**Comment by qdm12** [comment, marked as answer]
Please subscribe to #857 this is the same problem, and it's likely accepting the traffic coming from all detected interfaces will solve it. I'll comment on issue #857 once the change is made. I'll also add an environment variable later on to define a comma separated string to define from which interfaces you want to accept input traffic.
